clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -596,46 +612,38 @@ invariant_checker_t::print_results()
 }
 
 std::string
-invariant_checker_t::check_for_pc_discontinuity(per_shard_t *shard,
-                                                const memref_t &memref)
+invariant_checker_t::check_for_pc_discontinuity(
+    per_shard_t *shard, const memref_t &memref,
+    const std::unique_ptr<instr_t> &cur_instr_decoded, const bool expect_encoding)
 {
     std::string error_msg = "";
     bool have_cond_branch_target = false;
     addr_t cond_branch_target = 0;
-
     if (shard->prev_instr_.instr.addr != 0 /*first*/ &&
-        type_is_instr_direct_branch(shard->prev_instr_.instr.type) &&
         // We do not bother to support legacy traces without encodings.
-        TESTANY(OFFLINE_FILE_TYPE_ENCODINGS, shard->file_type_)) {
-        addr_t trace_pc = shard->prev_instr_.instr.addr;
+        expect_encoding && type_is_instr_direct_branch(shard->prev_instr_.instr.type)) {
         if (shard->prev_instr_.instr.encoding_is_new)
```

Comment:
nit: can declare a local var `prev_instr_trace_pc` for shard->prev_instr_.instr.addr since it's used multiple times here.

Reply:
Done.